### PR TITLE
fix(event_manager): Emit TRANSACTION outcomes if metrics are disabled

### PR DESCRIPTION
**Pull request by jan-auer**
In #40507 we started to count transaction metrics in the `transaction` data category and transaction events in the `transaction_indexed` data category. That PR missed that metrics extraction can be disabled, in which case the old behavior of counting events as `transaction` should remain. Relay already implemented this logic since getsentry/relay#1537 based on the metrics extraction flag.

This PR adds a feature check to the `organizations:transaction-metrics-extraction` feature, which is the same feature flag used to control Relay's behavior. We also remove the previously used option to sample a percentage of organizations into metrics extraction.

The default for this feature remains off (`False`) until metrics components have been added to all deployment targets including self-hosted.

**Note:** This requires that the above is set, and that `relay.transaction-metrics-org-sample-rate` is no longer used for this.

Requires getsentry/getsentry#8935
Fixes getsentry/team-ingest#1

### CODE REVIEW

**Review by jjbayer — APPROVED** [review]
We must deploy https://github.com/getsentry/getsentry/pull/8935 before we can merge this.

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -1036,11 +1036,10 @@ def SOCIAL_AUTH_DEFAULT_USERNAME():
...
     # Enable metric alert charts in email/slack
     "organizations:metric-alert-chartcuterie": False,
-    # Automatically extract metrics during ingestion.
-    #
-    # XXX(ja): DO NOT ENABLE UNTIL THIS NOTICE IS GONE. Relay experiences
-    # gradual slowdown when this is enabled for too many projects.
+    # Extract metrics for sessions during ingestion.
     "organizations:metrics-extraction": False,
+    # Extraction metrics for transactions during ingestion.
+    "organizations:transaction-metrics-extraction": False,
```
Comment (jan-auer) [inline src/sentry/conf/server.py:1042]: This was missing entirely. The default was implicitly `False`.

File: src/sentry/conf/server.py [hunk 2]
```diff
@@ -1036,11 +1036,10 @@ def SOCIAL_AUTH_DEFAULT_USERNAME():
     "organizations:metrics": False,
     # Enable metric alert charts in email/slack
     "organizations:metric-alert-chartcuterie": False,
-    # Automatically extract metrics during ingestion.
-    #
-    # XXX(ja): DO NOT ENABLE UNTIL THIS NOTICE IS GONE. Relay experiences
-    # gradual slowdown when this is enabled for too many projects.
+    # Extract metrics for sessions during ingestion.
```
Comment (jan-auer) [inline src/sentry/conf/server.py:1039]: The slowdown in Relay was resolved in https://github.com/getsentry/relay/pull/1065

File: src/sentry/conf/server.py (hunk 1 above)
Comment (markstory) [reply src/sentry/conf/server.py:1042]: Should we remove this flag's fallback to Flagr? Currently it has a `True` for the fallback.

Comment (jan-auer) [reply src/sentry/conf/server.py:1042]: 👍  It's removed with 3659729.

**Comment by jan-auer** [comment]
> We must deploy https://github.com/getsentry/getsentry/pull/8935 before we can merge this.

@jjbayer Both could have been deployed together. We have deployed in the meanwhile, though.

**Review by markstory — APPROVED** [review]

**Review by olksdr — APPROVED** [review]

**Comment by jjbayer** [comment]
> Both could have been deployed together. We have deployed in the meanwhile, though.

@jan-auer right, forgot that both repos are part of the same deploy for a moment.

**Pull request merged**